components/modal/index.js

```diff
@@ -0,0 +1,176 @@
+	 * separate div for this specific modal will be appended to that.
+	 */
+	prepareDOM() {
+		if ( ! parentElement ) {
+			parentElement = document.createElement( 'div' );
+			document.body.appendChild( parentElement );
+		}
+		this.node = document.createElement( 'div' );
+		parentElement.appendChild( this.node );
+	}
+
+	/**
+	 * Removes the specific mounting point for this modal from the DOM.
+	 */
+	cleanDOM() {
+		parentElement.removeChild( this.node );
+	}
+
+	/**
+	 * Appends the modal's node to the DOM, so the portal can render the
+	 * modal in it. Also calls the openFirstModal when this is the first modal to be
+	 * opened.
+	 */
+	componentDidMount() {
+		openModalCount++;
+
+		if ( openModalCount === 1 ) {
+			this.openFirstModal();
+		}
+	}
+
+	/**
+	 * Removes the modal's node from the DOM. Also calls closeLastModal when this is
+	 * the last modal to be closed.
+	 */
+	componentWillUnmount() {
+		openModalCount--;
+
+		if ( openModalCount === 0 ) {
+			this.closeLastModal();
+		}
+
+		this.cleanDOM();
+	}
+
+	/**
+	 * Prepares the DOM for this modal and any additional modal to be mounted.
+	 *
+	 * It appends an additional div to the body for the modals to be rendered in,
+	 * it hides any other elements from screen-readers and adds an additional class
+	 * to the body to prevent scrolling while the modal is open.
+	 */
+	openFirstModal() {
+		ariaHelper.hideApp( parentElement );
+		document.body.classList.add( this.props.bodyOpenClassName );
+	}
+
+	/**
+	 * Cleans up the DOM after the last modal is closed and makes the app available
+	 * for screen-readers again.
+	 */
+	closeLastModal() {
+		document.body.classList.remove( this.props.bodyOpenClassName );
+		ariaHelper.showApp();
+	}
+
+	/**
+	 * Renders the modal.
+	 *
+	 * @return {WPElement} The modal element.
+	 */
+	render() {
+		const {
+			overlayClassName,
+			className,
+			onRequestClose,
+			title,
+			icon,
+			closeButtonLabel,
+			children,
+			aria,
+			...otherProps
+		} = this.props;
+
+		return createPortal(
+			<div className={ classnames(
+				'components-modal__screen-overlay',
+				overlayClassName
+			) }>
+				<ModalFrame
+					className={ classnames(
+						'components-modal__frame',
+						className
+					) }
+					onRequestClose={ onRequestClose }
+					aria={ {
+						labelledby: this.headingId,
+						describedby: aria.describedby,
+					} }
+					{ ...otherProps } >
+					<ModalHeader
+						closeLabel={ closeButtonLabel }
+						onClose={ onRequestClose }
+						title={ title }
+						headingId={ this.headingId }
+						icon={ icon } />
+					<div
+						className={ 'components-modal__content' }>
+						{ children }
+					</div>
+				</ModalFrame>
+			</div>,
+			this.node
+		);
+	}
+}
+
+Modal.defaultProps = {
+	bodyOpenClassName: 'modal-open',
+	role: 'dialog',
+	title: null,
+	onRequestClose: noop,
+	focusOnMount: true,
+	shouldCloseOnEsc: true,
+	shouldCloseOnClickOutside: true,
+	style: {
```

Comment:
Can we just remove this prop? I personally don't care if react-modal has certain functionality, if we don't need it or it doesn't fit our criteria for quality (encouraging good practices, etc), it shouldn't be included.

Reply:
I already removed it, but forgot to remove it from the defaultProps.